src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -291,38 +293,39 @@ public void CreateValidationMetadata(ValidationMetadataProviderContext context)
             }
         }
 
-        // Return non-empty name specified in a [Display] attribute for a field, if any; field.Name otherwise.
-        private static string GetDisplayName(FieldInfo field)
+        // Return non-empty group specified in a [Display] attribute for a field, if any; string.Empty otherwise.
+        private static string GetDisplayGroup(FieldInfo field)
         {
             var display = field.GetCustomAttribute<DisplayAttribute>(inherit: false);
             if (display != null)
             {
-                // Note [Display(Name = "")] is allowed.
-                var name = display.GetName();
-                if (name != null)
+                // Note [Display(Group = "")] is allowed.
+                var group = display.GetGroupName();
+                if (group != null)
                 {
-                    return name;
+                    return group;
                 }
             }
 
-            return field.Name;
+            return string.Empty;
         }
 
-        // Return non-empty group specified in a [Display] attribute for a field, if any; string.Empty otherwise.
-        private static string GetDisplayGroup(FieldInfo field)
+        private static string GetDisplayName(FieldInfo fieldInfo, IStringLocalizer stringLocalizer)
```

Comment:
For the future: If we see this method on the containing `Func` show up in perf profiles, might be worthwhile for this method to return `Func<string>` instead. Most of the decision tree is known when the `Func` is created. Even `string.IsNullOrEmpty(name)` won't vary when `displayAttribute.ResourceType == null`.